src/engine/ult.c

```diff
@@ -329,7 +329,9 @@ sched_ult2xs(int xs_type, int tgt_id)
 {
 	uint32_t	xs_id;
 
-	D_ASSERT(tgt_id >= 0 && tgt_id < dss_tgt_nr);
+	if (xs_type != DSS_XS_SYS)
```

Comment:
either we revert this change and ensure caller always pass 0 tgt_id when the tgt_id is not used, or we change it to "if (xs_type == DSS_XS_VOS || xs_type == DSS_XS_OFFLOAD || xs_type == DSS_XS_IOFW)"